server/paymentProviders/thegivingblock/index.js

```diff
@@ -46,13 +47,18 @@ async function handleErrorsAndRetry(result, path, options = {}, account = null)
         }
         const response = await fetch(`${API_URL}${path}`, options);
         const result = await response.json();
+        if (result.data.errorMessage) {
+          throw new Error(`The Giving Block: ${result.data.errorMessage} ${result.data.meta.errorCode}`);
+        }
         return result.data;
       } catch (err) {
         error = err;
       }
-      throw error;
+      logger.error(error.message);
+      throw new Error(GENERIC_ERROR_MSG);
```

Comment:
I don't see the point having this out of the `catch`.

Reply:
Yes makes sense. I've put this within the catch and simplified. Thanks. 👍🏽